### ISSUE TITLE
Update PipeRPCPlugin compatibility to pytorch 1.7.0+

### ISSUE THREAD

**Issue by y-rokutan** [PyTorchLightning/pytorch-lightning#5293]
## 🚀 Feature

- Latest pytorch-lightning supports Sequential Model Parallelism, but compatible only with torch==1.6.0. Supporting 1.7.0+ allows us to exploit latest GPUs such as RTX 3000 series.

### Motivation

- Fine-tuning models too large for a single GPU.

### Pitch

- PipeRPCPlugin to support torch==1.7.0 or higher.

### Alternatives

- pl.Trainer supports `accelerator='sharded'` option without data-parallelism. (i.e. batch_size=1)

### Additional context
N/A

**Comment by tchaton** [comment]
Dear @y-rokutan,

Have you tried PipeRCPPluging. Did it work fine for you ?
I will try to see if it works with 1.7+ and keep you updated.

Best,
T.C

**Comment by y-rokutan** [comment]
Hi @tchaton.

I've tried PipeRCPPluging on torch==1.8.0.dev20201229+cu110 and pytorch-lightning==1.1.0 (compatible with RTX 3090),
but it shows `PipeRPCPlugin requires FairScale and currently is only supported on PyTorch 1.6.` error:
(it means `FAIRSCALE_AVAILABLE and LooseVersion(torch.__version__) == LooseVersion("1.6.0")` is False)

removing ` LooseVersion(torch.__version__) == LooseVersion("1.6.0")` allows run code without error, but the model doesn't split on GPUs correctly: only GPU_0 voltage/vRam jumps and others doesn't load the model.

BTW, I'm trying to train BERT-like transformer which is too big for a single GPU even if batch_size=1.
If I can set batch_size=1 (i.e. no data-parallelism) and accelerator='ddp_sharding' option for pl.Trainer, it would be a good walkaround.
Does anyone know how to enable sharding without dp?

**Comment by Borda** [comment]
@y-rokutan mind open a request on FairScale when they make an official release with wider version support?
(and pls link the request here to this issue)

**Comment by y-rokutan** [comment]
Hi @Borda,
To my understanding, fairscale [supports torch==1.7.1](https://github.com/facebookresearch/fairscale/pull/251) and [CUDA=11.0](https://github.com/facebookresearch/fairscale/issues/190), so fairscale seems not the cause of this issue.
What kind of test can I do to determine where the problem is?

**Comment by tchaton** [comment]
Hey @y-rokutan,

```
removing LooseVersion(torch.__version__) == LooseVersion("1.6.0") allows run code without error, but the model doesn't split on GPUs correctly: only GPU_0 voltage/vRam jumps and others doesn't load the model.
```
That's wierd. I look into this.

By the way `ddp_sharded` does perform model parallelism, but optimizer state and gradient sharding.

Best,
T.C

**Comment by SebastienEske** [comment]
Hello,

I am planning to have a few RTX 3090 so I am curious, why does pytorch 1.6 not support this card?

Also,
> By the way `ddp_sharded` does perform model parallelism, but optimizer state and gradient sharding.
Indeed but if the issue is the activations (as in computer vision applications) then it does not help.

**Comment by Borda** [comment]
> I am planning to have a few RTX 3090 so I am curious, why does pytorch 1.6 not support this card?

 think that you should ask them :]
so far I know the recommendation is to use PT 1.8
you could also have a question about py3.9 compatibility for older PT versions :]

**Comment by SebastienEske** [comment]
> > I am planning to have a few RTX 3090 so I am curious, why does pytorch 1.6 not support this card?
> 
> think that you should ask them :]
> so far I know the recommendation is to use PT 1.8
> you could also have a question about py3.9 compatibility for older PT versions :]

I figured it out: https://pytorch.org/docs/stable/notes/cuda.html#tf32-on-ampere
TensorFloat support is only available starting from pytorch 1.7+ and from the new generation of NVIDIA GPUs (Ampere)

**Comment by tchaton** [comment]
Dear @y-rokutan,

Closing this PR as SequentialPlugin will be deprecated.

Best,
T.C
